src/features/MentorPage/MentorListItems/MentorListItems.tsx

```diff
@@ -1,29 +1,47 @@
 import styled from 'styled-components';
 import ListCard from '../ListCard';
 import { MentorListItemsProps } from './types';
-import * as cssVariables from '../../../components/CommonTextStyles/variables';
+import * as cssVariables from '../../../components/variables';
+import { handleSetVisibleCardProps } from '../MentorPage';
+import { ListCardProps } from '../ListCard/types';
 
 /**
- * The logic for mentor cards shown in the Mentor page
- * An array with cards that need to be shown should be
- * passed to CardsElement where layout will be added.
+ * Mentor cards listing on mentor page
  */
 
+type Props = {
+  setVisibleCard: ({
+    shouldShowMentorCard,
+    mentorCardData,
+  }: handleSetVisibleCardProps) => void;
+  item: ListCardProps;
+};
+
+const MapMentorCard = ({ setVisibleCard, item }: Props) => {
+  if (item) {
+    return (
+      <ListCard
+        key={item.mentor.displayName}
+        mentorCardData={item}
+        setVisibleCard={setVisibleCard}
+      />
+    );
+  }
+  return <></>;
+};
+
 const MentorListItems = ({
   setVisibleCard,
-  mentorData,
+  mentorCardData,
 }: MentorListItemsProps) => {
-  return (
-    <CardsList data-testid="mentor-cards-container">
-      {mentorData.map(item => (
-        <ListCard
-          key={item.mentor.displayName}
-          mentorData={item}
-          setVisibleCard={setVisibleCard}
-        />
-      ))}
-    </CardsList>
-  );
+  if (mentorCardData !== undefined) {
+    return (
+      <CardsList data-testid="mentor-cards-container">
+        {mentorCardData.map(item => MapMentorCard({ setVisibleCard, item }))}
+      </CardsList>
+    );
+  }
+  return <></>;
```

Comment:
make it required

Reply:
P.S. This whole file MentorListItems.tsx could be simplified to this =>

![image](https://user-images.githubusercontent.com/34128180/183894547-889743ee-b853-4a94-a0e5-601ede0d2cde.png)

we get rid of the mapMentorData-renderFunction and no need to check the data if it is undefined =P